### PR TITLE
Documentation for On Premise Deployment with External Managed Services

### PR DESCRIPTION
**Pull request by Sathvikreddy14**
Signed-off-by: Sathvikreddy14 <creddy@progress.com>

### :nut_and_bolt: Description: What code changed, and why?
Added documentation steps for setting up On Premise deployment with AWS Managed Services and On Premise deployment with Self managed services.

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [ ] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [ ] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [ ] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [ ] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [ ] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md [hunk 1]
```diff
@@ -198,7 +198,7 @@ opensearch_private_ips = ["A1.A2.A3.A4","B1.B2.B3.B4","C1.C2.C3.C4"]
 postgresql_private_ips = ["D1.D2.D3.D4","E1.E2.E3.E4","F1.F2.F3.F4"]
 ```
 
-### Minimum changes to be made
+### Minimum changes to be made for On Premise Deployment
```
Comment (ArvinthC3000) [inline components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md:201]:
```suggestion
### Minimum changes to be made for On-Premise Deployment
```

File: components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md [hunk 2]
```diff
@@ -208,10 +208,173 @@ postgresql_private_ips = ["D1.D2.D3.D4","E1.E2.E3.E4","F1.F2.F3.F4"]
 - `opensearch_private_ips` Eg: ["192.0.2.1", "192.0.2.2", "192.0.2.2"]
 - `postgresql_private_ips` Eg: ["192.0.3.1", "192.0.3.2", "192.0.3.2"]
 
+## On Premise Setup with AWS Managed Services.
```
Comment (ArvinthC3000) [inline components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md:211]:
```suggestion
## On-Premise Setup with AWS Managed Services.
```

File: components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md [hunk 3]
```diff
@@ -208,10 +208,173 @@ postgresql_private_ips = ["D1.D2.D3.D4","E1.E2.E3.E4","F1.F2.F3.F4"]
...
+
+### Prerequisites
+
+- Follow the Prerequisites for On Premise deployment. Click [here](https://docs.chef.io/automate/ha_onprim_deployment_procedure/#prerequisites).
```
Comment (ArvinthC3000) [inline components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md:215]: Chang `On Premise` to 'On-Premise' wherever applicable

File: components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md [hunk 4]
```diff
@@ -208,10 +208,173 @@ postgresql_private_ips = ["D1.D2.D3.D4","E1.E2.E3.E4","F1.F2.F3.F4"]
...
+- You can exclude the creation of instances for OpenSearch and Postgresql as you will be using AWS OpenSearch and AWS RDS Postgresql.
+- Setup AWS RDS Postgresql 13.5. Click [here](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_GettingStarted.CreatingConnecting.PostgreSQL.html) to know more. Make sure to open the required port in Security Groups while creating AWS RDS Postgresql.
+- Setup AWS OpenSearch 1.2. Click [here](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/createupdatedomains.html) to know more.
+- For Backup and Restore with Managed Service. Click [here](/automate/managed_services/#prerequisites) to know more.
+- Create the Virtual Private Cloud (VPC) in AWS before starting or using default. Click [here](/automate/ha_vpc_setup/) to know more about VPC and CIDR creation.
+- Get AWS credetials (`aws_access_key_id` and `aws_secret_access_key`) which have privileges like: `AmazonS3FullAccess`, `AdministratorAccess`, `AmazonAPIGatewayAdministrator`. Click [here](/automate/ha_iam_user/) to know more on how to create IAM Users.
```
Comment (punitmundra) [inline components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md:221]: why this(AmazonAPIGatewayAdministrator) is required ?

File: components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md [hunk 5]
```diff
@@ -208,10 +208,173 @@ postgresql_private_ips = ["D1.D2.D3.D4","E1.E2.E3.E4","F1.F2.F3.F4"]
 - `opensearch_private_ips` Eg: ["192.0.2.1", "192.0.2.2", "192.0.2.2"]
 - `postgresql_private_ips` Eg: ["192.0.3.1", "192.0.3.2", "192.0.3.2"]
 
+## On-Premise Setup with AWS Managed Services.
+
+### Prerequisites
+
+- Follow the Prerequisites for On-Premise deployment. Click [here](https://docs.chef.io/automate/ha_onprim_deployment_procedure/#prerequisites).
+- You can exclude the creation of instances for OpenSearch and Postgresql as you will be using AWS OpenSearch and AWS RDS Postgresql.
```
Comment (punitmundra) [inline components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md:216]: reframe the sentence. This deployment excludes the installation for PG and OS as we are using the AWS Managed services.

**Review by punitmundra — CHANGES_REQUESTED** [review]

File: components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md [hunk 6]
```diff
@@ -208,10 +208,173 @@ postgresql_private_ips = ["D1.D2.D3.D4","E1.E2.E3.E4","F1.F2.F3.F4"]
...
+- This deployment excludes the installation for Postgresql and OpenSearch as we are using the Self Managed services.
+
+See the steps [here](https://docs.chef.io/automate/ha_onprim_deployment_procedure/#run-these-steps-on-bastion-host-machine) to run on Bastion to download latest Automate CLI and Airgapped Bundle.
+
+Update Config with relevant data. Click [here](/automate/ha_onprim_deployment_procedure/#sample-config-to-setup-on-premise-deployment-with-self-managed-services) for sample config for Self Managed Services.
+
+- Set Self Managed Config Details:
+    - Provide instance count as `0` for [opensearch.config] and [postgresql.config] and leave the values of opensearch_private_ips and postgresql_private_ips as empty.
+    - Set `type` as `self-managed`, As these deployment steps are for Managed Services AWS Deployment. The default value is blank, which should be changed.
+    - Set `instance_url`, `superuser_username`, `superuser_password`, `dbuser_username`, `dbuser_password` of your Self Managed RDS.
```
Comment (jayvikramsharma1) [inline components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md:309]: for your Self Managed RDS.

File: components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md [hunk 7]
```diff
@@ -208,10 +208,173 @@ postgresql_private_ips = ["D1.D2.D3.D4","E1.E2.E3.E4","F1.F2.F3.F4"]
...
+    - Set `instance_url` as the URL with Port No. For example: `"10.1.2.189:7432"`.
+    - Provide the Root ca value of Postgresql `postgresql_root_cert`.
+    - Set `opensearch_domain_name`, `opensearch_domain_url`, `opensearch_username`, `opensearch_user_password` of your Self Managed OpenSearch.
```
Comment (jayvikramsharma1) [inline components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md:312]: for your Self Managed OpenSearch.

File: components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md [hunk 8]
```diff
@@ -208,10 +208,173 @@ postgresql_private_ips = ["D1.D2.D3.D4","E1.E2.E3.E4","F1.F2.F3.F4"]
...
+
+Update Config with relevant data. Click [here](/automate/ha_onprim_deployment_procedure/#sample-config-to-setup-on-premise-deployment-with-aws-managed-services) for sample config of AWS Managed Services.
+
+- Set AWS Config Details:
+    - Provide instance count as `0` for [opensearch.config] and [postgresql.config] and leave the values of opensearch_private_ips and postgresql_private_ips as empty.
+    - Set `type` as `aws`, As these deployment steps are for Managed Services AWS Deployment. The default value is blank, which should be changed.
+    - Set `instance_url`, `superuser_username`, `superuser_password`, `dbuser_username`, `dbuser_password` from the **Managed AWS RDS Postgresql** created in the Prerequisite steps.
+    - Set `instance_url` as the URL with Port No. For example: `"database-1.c2kvay.eu-north-1.rds.amazonaws.com:5432"`
+    - Set `opensearch_domain_name`, `opensearch_domain_url`, `opensearch_username`, `opensearch_user_password` from the **Managed AWS OpenSearch** created in the Prerequisite steps.
+    - Set `opensearch_domain_url` as the URL without Port No. For example: `"vpc-automate-ha-cbyqy5q.eu-north-1.es.amazonaws.com"`.
```
[the remark this answers is not in the document]
Comment (Sathvikreddy14) [reply components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md:233]: In case of AWS Managed Services, port no is not required

**Review by jayvikramsharma1 — APPROVED** [review]

File: components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md [hunk 9]
```diff
@@ -208,10 +208,173 @@ postgresql_private_ips = ["D1.D2.D3.D4","E1.E2.E3.E4","F1.F2.F3.F4"]
...
+## On-Premise Setup with AWS Managed Services.
+
+### Prerequisites
+
+- Follow the Prerequisites for On-Premise deployment. Click [here](https://docs.chef.io/automate/ha_onprim_deployment_procedure/#prerequisites).
+- This deployment excludes the installation for Postgresql and OpenSearch as we are using the AWS Managed services.
+- Setup AWS RDS Postgresql 13.5. Click [here](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_GettingStarted.CreatingConnecting.PostgreSQL.html) to know more. Make sure to open the required port in Security Groups while creating AWS RDS Postgresql.
+- Setup AWS OpenSearch 1.2. Click [here](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/createupdatedomains.html) to know more.
+- For Backup and Restore with Managed Service. Click [here](/automate/managed_services/#prerequisites) to know more.
+- Create the Virtual Private Cloud (VPC) in AWS before starting or using default. Click [here](/automate/ha_vpc_setup/) to know more about VPC and CIDR creation.
```
Comment (ArvinthC3000) [inline components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md:220]: Create the Virtual Private Cloud (VPC) in AWS or use the default one

File: components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md [hunk 10]
```diff
@@ -208,10 +208,173 @@ postgresql_private_ips = ["D1.D2.D3.D4","E1.E2.E3.E4","F1.F2.F3.F4"]
...
+- Get AWS credetials (`aws_access_key_id` and `aws_secret_access_key`) which have privileges like: `AmazonS3FullAccess`, `AdministratorAccess`. Click [here](/automate/ha_iam_user/) to know more on how to create IAM Users.
```
Comment (ArvinthC3000) [inline components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md:221]:
```suggestion
- Get AWS credentials (`aws_access_key_id` and `aws_secret_access_key`) which have privileges that includes: `AmazonS3FullAccess`, and `AdministratorAccess`. Click [here](/automate/ha_iam_user/) to know more about how to create IAM Users.
```

File: components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md [hunk 11]
```diff
@@ -208,10 +208,173 @@ postgresql_private_ips = ["D1.D2.D3.D4","E1.E2.E3.E4","F1.F2.F3.F4"]
...
+
+See the steps [here](https://docs.chef.io/automate/ha_onprim_deployment_procedure/#run-these-steps-on-bastion-host-machine) to run on Bastion to download latest Automate CLI and Airgapped Bundle.
+
+Update Config with relevant data. Click [here](/automate/ha_onprim_deployment_procedure/#sample-config-to-setup-on-premise-deployment-with-aws-managed-services) for sample config of AWS Managed Services.
+
+- Set AWS Config Details:
+    - Provide instance count as `0` for [opensearch.config] and [postgresql.config] and leave the values of opensearch_private_ips and postgresql_private_ips as empty.
```
Comment (ArvinthC3000) [inline components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md:228]:
```suggestion
    - Provide instance count as `0` for both [opensearch.config] and [postgresql.config] and leave the values of opensearch_private_ips and postgresql_private_ips empty array.
```

File: components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md [hunk 12]
```diff
@@ -208,10 +208,173 @@ postgresql_private_ips = ["D1.D2.D3.D4","E1.E2.E3.E4","F1.F2.F3.F4"]
...
+    - Set `type` as `aws`, As these deployment steps are for Managed Services AWS Deployment. The default value is blank, which should be changed.
```
Comment (ArvinthC3000) [inline components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md:229]:
```suggestion
    - Set `type` as `aws`, as these deployment steps are for Managed Services AWS Deployment. The default value is blank.
```

File: components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md [hunk 13]
```diff
@@ -208,10 +208,173 @@ postgresql_private_ips = ["D1.D2.D3.D4","E1.E2.E3.E4","F1.F2.F3.F4"]
...
+# eg: os_snapshot_user_access_key_secret = "skP4Mqihj....................anAXAX"
+os_snapshot_user_access_key_secret = ""
+```
+
+## On-Premise Setup with Self Managed Services
+
+### Prerequisites
+
+- Follow the Prerequisites for On-Premise deployment. Click [here](https://docs.chef.io/automate/ha_onprim_deployment_procedure/#prerequisites).
+- This deployment excludes the installation for Postgresql and OpenSearch as we are using the Self Managed services.
```
Comment (ArvinthC3000) [inline components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md:300]: Make it a note

File: components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md [hunk 14]
```diff
@@ -208,10 +208,173 @@ postgresql_private_ips = ["D1.D2.D3.D4","E1.E2.E3.E4","F1.F2.F3.F4"]
...
+
+See the steps [here](https://docs.chef.io/automate/ha_onprim_deployment_procedure/#run-these-steps-on-bastion-host-machine) to run on Bastion to download latest Automate CLI and Airgapped Bundle.
+
+Update Config with relevant data. Click [here](/automate/ha_onprim_deployment_procedure/#sample-config-to-setup-on-premise-deployment-with-self-managed-services) for sample config for Self Managed Services.
+
+- Set Self Managed Config Details:
+    - Provide instance count as `0` for [opensearch.config] and [postgresql.config] and leave the values of opensearch_private_ips and postgresql_private_ips as empty.
```
Comment (ArvinthC3000) [inline components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md:307]:
```suggestion
    - Provide instance count as `0` for both [opensearch.config] and [postgresql.config] and leave the values of opensearch_private_ips and postgresql_private_ips empty array.
```

File: components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md [hunk 15]
```diff
@@ -208,10 +208,173 @@ postgresql_private_ips = ["D1.D2.D3.D4","E1.E2.E3.E4","F1.F2.F3.F4"]
...
+    - Set `type` as `self-managed`, As these deployment steps are for Managed Services AWS Deployment. The default value is blank, which should be changed.
```
Comment (ArvinthC3000) [inline components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md:308]:
```suggestion
    - Set `type` as `self-managed`, As these deployment steps are for Managed Services AWS Deployment. The default value is blank.
```

**Review by punitmundra — APPROVED** [review]

File: components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md [hunk 16]
```diff
@@ -208,10 +208,173 @@ postgresql_private_ips = ["D1.D2.D3.D4","E1.E2.E3.E4","F1.F2.F3.F4"]
...
+os_snapshot_user_access_key_id = ""
+# eg: os_snapshot_user_access_key_secret = "skP4Mqihj....................anAXAX"
+os_snapshot_user_access_key_secret = ""
+```
 ### How To Add More Nodes to the On-Prem Deployment
 
 - Open the `config.toml` at bastion node
-- change the `instance_count` value, as explained in below example.
+- Change the `instance_count` value, as explained in below example.
 - Add the `Ip address` for the respetive node at the end, as explained in the example.
```
Comment (ArvinthC3000) [inline components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md:378]:
```suggestion
- Add the IP address for the respective node at the end, as explained in the example.
```

File: components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md [hunk 17]
```diff
@@ -208,10 +208,173 @@ postgresql_private_ips = ["D1.D2.D3.D4","E1.E2.E3.E4","F1.F2.F3.F4"]
...
 - Make sure that all the necesssary port and fire wall setting are align in the new node.
```
Comment (ArvinthC3000) [inline components/docs-chef-io/content/automate/ha_onprim_deployment_procedure.md:379]:
```suggestion
- Make sure that all the necessary ports and firewall settings are aligned in the new node.
```

**Review by dishanktiwari2501 — APPROVED** [review]

**Pull request merged**